lib/bitcoin/script/script_parser.rb

```diff
@@ -0,0 +1,114 @@
+module Bitcoin
+  class Script
+    class ScriptParser < Struct.new(:bytes, :offset)
```

Comment:
Why not just add `attr_accessor` in the class? Inheriting from Struct seems unnecessarily confusing.